python_modules/dagster/dagster/_utils/caching_instance_queryer.py

```diff
@@ -186,10 +186,21 @@ def get_in_progress_asset_subset(
             # be launched, and then run B completes before run A. In these cases, the computation
             # below will consider the asset to not be in progress, as the latest planned event
             # will be associated with a completed run.
+            asset_record = self.get_asset_record(asset_key)
+            last_materialized_run_id = (
+                asset_record.asset_entry.last_materialization_record.run_id
+                if asset_record and asset_record.asset_entry.last_materialization_record
+                else None
+            )
             planned_materialization_info = (
                 self.instance.event_log_storage.get_latest_planned_materialization_info(asset_key)
             )
```

Comment:
as long as we are already fetching the asset record, can we check event_log_storage.asset_records_have_last_planned_materialization_storage_id and pull that run ID from the asset record in the event that it is true?

might make testing a little trickier but brings each call of this from two queries to one query